File: 160. Intersection of Two Linked Lists/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func getIntersectionNode(headA, headB *ListNode) *ListNode {
    var curA *ListNode = headA
    var curB *ListNode = headB
    var intersectedNode *ListNode
    if curA == nil || curB == nil {
        return nil
    } 
    flagA := false
    flagB := false
    for {
        if curA == curB {
            intersectedNode = curA
            break
        }
        if flagA && flagB && curA == nil && curB == nil {
            break
        }
        if curA == nil {
            curA = headB
            flagA = true
            continue    // important
        }
        if curB == nil {
            curB = headA
            flagB = true
            continue    // important
        }
        curA = curA.Next
        curB = curB.Next
    }
    return intersectedNode
}

File: 160. Intersection of Two Linked Lists/solution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var getIntersectionNode = function(headA, headB) {
    let cur1 = headA
    let cur2 = headB
    let f1 = false
    let f2 = false
    while(cur1 && cur2) {
        if(cur1 === cur2) {
            return cur1
        }
        cur1 = cur1.next
        cur2 = cur2.next
        if(cur1 === null && !f1) {
            f1 = true
            cur1 = headB
        }
        if(cur2 === null && !f2) {
            f2 = true
            cur2 = headA
        }
    }
    return null
};